pylint/checkers/utils.py

```diff
@@ -1831,7 +1831,11 @@ def is_sys_guard(node: nodes.If) -> bool:
             and value.as_string() == "sys.version_info"
         ):
             return True
-
+    elif isinstance(node.test, nodes.Attribute) and node.test.as_string() in {
```

Comment:
I think we should be using `qname()` here instead of `as_string()`.

Reply:
Thank you, this seems like a good idea. Unless I misunderstood something, it's not so easy because `six.PY2` or `six.PY3` cannot be infered.

If I try something like this:

```py
import astroid

six_py2 = astroid.extract_node('''
    import six
    six.PY2 #@
''')
print(six_py2.inferred())
```

this is because [`six.PY2`](https://github.com/benjaminp/six/blob/c1b416f24de52ebd8eaed3e379ab85ec9c3e935b/six.py#L36) is defined as 

```py
PY2 = sys.version_info[0] == 2
```

such a `Compare` node is uninferable, because as described in https://github.com/pylint-dev/astroid/blob/0823ae58cf0d2d8fa62122ec2bab1e73c70dba53/astroid/nodes/node_classes.py#L1872-L1873 "if any item is uninferable ... return Uninferable". In this code, `sys.version_info` is uninferable.

One way could be to make the `six` brain in astroid understands `six.PY2` and `six.PY3`, the change would be something like this ( applied [here](https://github.com/pylint-dev/astroid/blob/0823ae58cf0d2d8fa62122ec2bab1e73c70dba53/astroid/brain/brain_six.py#L115) - maybe slightly different because this is an extender for `six.moves` ):

```patch
diff --git a/astroid/brain/brain_six.py b/astroid/brain/brain_six.py
index c222a422..bfc9f8cc 100644
--- a/astroid/brain/brain_six.py
+++ b/astroid/brain/brain_six.py
@@ -112,6 +112,8 @@ def six_moves_transform():
     class Moves(object):
     {}
     moves = Moves()
+    PY2 = False
+    PY3 = True
     """
     ).format(_indent(_IMPORTS, "    "))
     module = AstroidBuilder(AstroidManager()).string_build(code)
```

Another idea that I did not try could be to add an astroid brain for `sys`, that would understand that `sys.version_info` is a namedtuple. I can maybe try this later, unless you think this is not a good idea.

Reply:
Your explanation is very clear. Let's go ahead and merge this as it is already an improvement. People should just not import `as five` :smile:

Reply:
Thanks, it would have been good to infer, I hope nobody imports `as five` ;) but this would have supported `from six import PY2`, which seems to be used a bit ( according to  https://github.com/search?q=%22from+six+import+PY2%22&type=code )